From 2a07df8b05767306c7a1088b05407f89a51e0646 Mon Sep 17 00:00:00 2001
From: Igor Mammedov <imammedo@redhat.com>
Date: Wed, 17 Sep 2014 12:52:27 -0400
Subject: [acpi] processor: do not mark present at boot but not onlined CPU as onlined

Message-id: <1410958350-13432-2-git-send-email-imammedo@redhat.com>
Patchwork-id: 95864
O-Subject: [RHEL7.1 PATCH 1/4] ACPI / processor: do not mark present at boot but not onlined CPU as onlined
Bugzilla: 968147
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=968147
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7981610
Upstream: 0b9d46dd7debf8e6dc8614106f1c1909fa8de64d

acpi_processor_add() assumes that present at boot CPUs
are always onlined, it is not so if a CPU failed to become
onlined. As result acpi_processor_add() will mark such CPU
device as onlined in sysfs and following attempts to
online/offline it using /sys/device/system/cpu/cpuX/online
attribute will fail.

Do not poke into device internals in acpi_processor_add()
and touch "struct device { .offline }" attribute, since
for CPUs onlined at boot it's set by:
  topology_init() -> arch_register_cpu() -> register_cpu()
before ACPI device tree is parsed, and for hotplugged
CPUs it's set when userspace onlines CPU via sysfs.

Signed-off-by: Igor Mammedov <imammedo@redhat.com>
Acked-by: Toshi Kani <toshi.kani@hp.com>
Cc: 3.11+ <stable@vger.kernel.org> # 3.11+
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/acpi_processor.c b/drivers/acpi/acpi_processor.c
index 5e013c9..7c5c312 100644
--- a/drivers/acpi/acpi_processor.c
+++ b/drivers/acpi/acpi_processor.c
@@ -405,7 +405,6 @@ static int acpi_processor_add(struct acpi_device *device,
   goto err;
 
  pr->dev = dev;
- dev->offline = pr->flags.need_hotplug_init;
 
  /* Trigger the processor driver's .probe() if present. */
  if (device_attach(dev) >= 0)
-- 
1.7.1